lib/models/repository-states/present.js

```diff
@@ -39,20 +39,46 @@ export default class Present extends State {
     this.operationStates = new OperationStates({didUpdate: this.didUpdate.bind(this)});
 
     this.commitMessage = '';
+    this.commitMessageTemplate = null;
+    this.fetchInitialMessage();
```

Comment:
I wonder if the commit message, commit message template, and merge message tracking belong more naturally in `Repository` rather than `Present`. Most methods in `Present` are intended to call a single git command with `GitShellOutStrategy`, manipulate the cache, and translate input and output to our model objects, while `Repository` methods implement higher-order composite behavior. Commit message derivation feels like a "composite" behavior to me 🤔 

With that said: we already have a bunch of behavior in `Repository` and `Present` that blurs these lines, and it would not be trivial to move all of this there. So maybe we could leave this as a future refactoring as part of a broader effort to make the division among those abstraction layers clear.

Reply:
cool - should I open an issue to refactor this in the future?  I'm on board with cleaning it up.